_example/db/pagination.go

```diff
@@ -24,6 +25,10 @@ type Pagination struct {
 }
 
 func (p *Pagination) Paginate(c *gin.Context) (*gorm.DB, error) {
+	if p == nil {
+		return nil, errors.New("Pagination struct got nil.")
+	}
```

Comment:
In golang, method of pointer receiver can be called from any nil pointer variable.
So, if some method that does not access the field, it is realistically possible to call the method that remains nil pointer.

I implemented this in view of such a use in golang. How do you think?

ref
- http://qiita.com/stsn/items/73714caf8458b1d973f2
- http://www.yunabe.jp/docs/golang_pitfall.html#this-selfnil